autoscheduler/frontend/src/components/SchedulingPage/SchedulePreview/SchedulePreview.tsx

```diff
@@ -1,19 +1,88 @@
 import * as React from 'react';
-import { useSelector } from 'react-redux';
+import * as Cookies from 'js-cookie';
+import { useDispatch, useSelector } from 'react-redux';
 import { List } from '@material-ui/core';
 import GenericCard from '../../GenericCard/GenericCard';
 import { RootState } from '../../../redux/reducer';
 import * as styles from './SchedulePreview.css';
 import ScheduleListItem from './ScheduleListItem/ScheduleListItem';
 import Schedule from '../../../types/Schedule';
+import { setSchedules } from '../../../redux/actions/schedules';
+import createThrottleFunction from '../../../utils/createThrottleFunction';
+import { parseAllMeetings } from '../../../redux/actions/courseCards';
 
-const SchedulePreview: React.FC = () => {
+const throttle = createThrottleFunction();
+
+interface SchedulePreviewProps {
+  // Is a prop so we can set the throttle time to a really low number when testing
+  throttleTime?: number;
+}
+
+const SchedulePreview: React.FC<SchedulePreviewProps> = ({ throttleTime = 5000 }) => {
+  const dispatch = useDispatch();
   const schedules = useSelector<RootState, Schedule[]>((state) => state.schedules);
+  const term = useSelector<RootState, string>((state) => state.term);
 
   const scheduleListItems = schedules.length === 0
     ? <p className={styles.noSchedules}>No schedules available.</p>
     : schedules.map((schedule, idx) => <ScheduleListItem index={idx} key={schedule.name} />);
 
+  // Whenever the term changes, fetch and load saved schedules
+  React.useEffect(() => {
+    if (term) {
+      fetch(`sessions/get_saved_schedules?term=${term}`).then(
+        (res) => res.json(),
+      ).then((arr: any[]): Schedule[] => arr.map((val) => ({
+        // TODO: Should we rename the schedule names if they're in the form of "Schedule #"?
+        name: val.name,
+        meetings: parseAllMeetings(val.sections),
+        saved: true,
+      }))).then((savedSchedules: Schedule[]) => {
+        dispatch(setSchedules(savedSchedules));
+      });
+    }
+
+    // on unmount, clear schedules
+    return (): void => {
+      dispatch(setSchedules([]));
+    };
+  }, [term, dispatch]);
+
+  // Call throttle to serialize and save the schedules anytime we make a change to the schedules
+  React.useEffect(() => {
+    if (!term) return;
+
+    // Serialize schedules and make API call
+    const saveSchedules = (): void => {
+      let serializedSchedules: { name: string; sections: number[] }[] = [];
+
+      if (schedules) {
```

Comment:
What is the difference between the case where `schedules` is `null` and the case where `schedules` is an empty array? The first would lead to making a fetch call with an empty array of schedules, and the first would just return. Why don't both have the same behavior?

Reply:
Had a reason at some point for having this check but I don't remember what it was so removing it